src/SetupApi.Desktop/SetupApi.cs

```diff
@@ -0,0 +1,195 @@
+        /// A pointer to a device information set that contains the device interfaces for which to
+        /// return information. This handle is typically returned by <see cref="SetupDiGetClassDevs(IntPtr,string,IntPtr,GetClassDevsFlags)" />.
+        /// </param>
+        /// <param name="deviceInterfaceData">
+        /// A pointer to an <see cref="DeviceInterfaceData" /> structure that specifies the
+        /// interface in DeviceInfoSet for which to retrieve details. A pointer of this type is typically returned by
+        /// <see cref="SetupDiEnumDeviceInterfaces(SafeDeviceInfoSetHandle,IntPtr,ref Guid,uint,ref DeviceInterfaceData)" />.
+        /// </param>
+        /// <param name="deviceInterfaceDetailData">
+        /// A pointer to an SP_DEVICE_INTERFACE_DETAIL_DATA structure to receive
+        /// information about the specified interface. This parameter is optional and can be <see langword="null" />. This
+        /// parameter must be <see langword="null" /> if DeviceInterfaceDetailSize is zero. If this parameter is specified, the
+        /// caller must set DeviceInterfaceDetailData.cbSize to sizeof(SP_DEVICE_INTERFACE_DETAIL_DATA) before calling this
+        /// function. The cbSize member always contains the size of the fixed part of the data structure, not a size reflecting the
+        /// variable-length string at the end.
+        /// </param>
+        /// <param name="deviceInterfaceDetailDataSize">
+        /// The size of the <paramref name="deviceInterfaceDetailData" /> buffer.
+        /// <para>This parameter must be zero if <paramref name="deviceInterfaceDetailData" /> is <see langword="null" />.</para>
+        /// </param>
+        /// <param name="requiredSize">
+        /// A pointer to a variable of type <see cref="uint" /> that receives the required size of the
+        /// DeviceInterfaceDetailData buffer. This size includes the size of the fixed part of the structure plus the number of
+        /// bytes required for the variable-length device path string. This parameter is optional and can be
+        /// <see langword="null" />.
+        /// </param>
+        /// <param name="deviceInfoData">
+        /// A pointer to a buffer that receives information about the device that supports the requested interface. The caller
+        /// must set <see cref="DeviceInfoData.Size" /> before calling this function either manually or via
+        /// <see cref="DeviceInfoData.Create" />.
+        /// <para>This parameter is optional and can be <see langword="null" />.</para>
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true" /> if the function completed without error. If the function completed with an
+        /// error, <see langword="false" /> is returned and the error code for the failure can be retrieved by calling
+        /// <see cref="Marshal.GetLastWin32Error" />.
+        /// </returns>
+        [DllImport(nameof(SetupApi), SetLastError = true, CharSet = CharSet.Auto)]
```

Comment:
Do you happen to be able to explain to me what `CharSet.Auto` means? .NET is fundamentally unicode, so when would it ever use Ansi? Or is it auto in the sense that it somehow magically figures out whether the native method expects ansi or not? Is this where .NET will add an A or W suffix to the method?